customisation.nix

```diff
@@ -209,7 +209,22 @@ rec {
     callPackageWith :: AttrSet -> ((AttrSet -> a) | Path) -> AttrSet -> a
     ```
   */
-  callPackageWith = autoArgs: fn: args:
+  callPackageWith = callGenericWith "lib.customisation.callPackageWith" makeOverridable;
+
+
+  /**
+    Very similar to callPackageWith, with the exception that it doesn't wrap the
+    package function in makeOverridable which adds `.override` and `.overrideDerivation`.
+    This is intended to be use when the calling function doesn't return a derivation.
+  */
+  scopeImportWith = callGenericWith "lib.customisation.scopeImportWith" lib.id;
+
+
+  /**
+    Not intended to be used directly. Instead use callPackageWith or scopeImportWith
+    dependening on whether you're importing packages or non-package expressions.
+  */
+  callGenericWith = funcName: continuation: autoArgs: fn: args:
```

Comment:
This should go in a let clause, to avoid it being available in the `customization` attr set. However, for the sake of the diff, just keep it here